pennylane/_qubit_device.py

```diff
@@ -114,6 +119,14 @@ def __init__(self, wires=1, shots=1000, analytic=True):
         """None or int: stores the hash of the circuit from the last execution which
         can be used by devices in :meth:`apply` for parametric compilation."""
 
+        self._cache = cache
```

Comment:
(Minor suggestion): could the name `_num_cached` or something more descriptive work here? Slightly confused that `self._cache` is not a boolean